cpp/include/cudf/functions.h

```diff
@@ -611,6 +611,69 @@ gdf_error gdf_extract_datetime_second(gdf_column *input, gdf_column *output);
 
 /* binary operators */
 
+/**
+ * @brief Binary operation function between gdf_scalar and gdf_column structs.
+ *
+ * The function performs the binary operation of a gdf_scalar operand and a
+ * gdf_column operand.
+ *
+ * The valid field in the gdf_column output will be 1 (by bit) when the two
+ * operands (lhs and rhs) are not null. Otherwise, it will be 0 (by bit).
+ *
+ * It is required to set in an appropriate manner the fields in the gdf_scalar and
```

Comment:
I think we also need to mention that the the inputs will be cast to the type specified in the output column.

Reply:
Agree with the first comment. I don't think we should mention the casting of input columns because

- It is an implementation detail that the user shouldn't be concerned about
- We aren't making any changes to the input objects and we don't want to hint that we are
- That is not true for every binary op. e.g. we don't do this for comparisons (>, ==, etc). For truediv and pow, we cast to double regardless of the out type.

Reply:
All fair points. 

I think the important point is to ensure that we document that the user of the API can control the output type via the `dtype` in the output column. So feel free to document that as you see fit.

Reply:
alrighty